src/modules/EnsonoBuild/exported/Build-DockerImage.Tests.ps1

```diff
@@ -207,6 +207,27 @@ Describe "Build-DockerImage" {
             # Ensure that the image is pused to the registry
             $Session.commands.list[2] | Should -BeLike "*docker* push docker.io/pester-tests:unittests"
         }
+
+        It "will build the image but not push if the NO_ENV variable has been set" {
+
+            # Set the environment variable to prevent the push
+            $env:NO_PUSH = "do not push"
+
+            # Call the function under test
+            Build-DockerImage -provider "generic" -name pester-tests -tag "unittests" -registry "docker.io" -push
+
+            # There should only be one command
+            $Session.commands.list.length | Should -Be 1
+
+            # Check the build command
+            $Session.commands.list[0] | Should -BeLike "*docker* build . -t pester-tests:unittests -t docker.io/pester-tests:unittests"
+
+            # Remove the environment variable
+            remove-item -path env:\NO_PUSH
```

Comment:
If the user had this set before the tests it will remove it from their environment if pester is run from within the same shell?

You may want to capture the value before setting it, and set it back after to what it was, not sure.

Also, if someone has this var set before running tests it'll use it for the other tests too which might cause them to fail.

Reply:
The problem we are going to have is that other tests are not going to pass if the user _does_ have the env var set as that will change the outcome of previous tests
I can check for the existence of it and save to a temporary variable, but I am not sure that it achieves anything.